newdevice: Added preliminary support for Ceragon FA2000 radios.

tamikkelsen opened the pull request:
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

It uses CeragonNet mib file.

This recognize Ceragon FA2000C,D radios, It also graphs signal strength. 
[CeragonNet.txt](https://github.com/librenms/librenms/files/980924/CeragonNet.txt)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/librenms/librenms?pullRequest=6601) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/librenms/librenms?pullRequest=6601) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/librenms/librenms?pullRequest=6601) it.</sub>

laf commented on includes/definitions/ceragon.yaml:
```diff
@@ -0,0 +1,16 @@
+os: ceragon
```
This is too generic for an os name.

I can't find much on the FA2000, is this the only product in the range? What OS / Firmware name does it have on the device?

laf replied:
Based on your mib name, is fibreair not a more appropriate os name?

laf commented on includes/discovery/sensors/signal/ceragon.inc.php:
```diff
@@ -0,0 +1,20 @@
+$divisor = 1;
+$type    = 'ceragon';
+
+$oids = snmp_walk($device, 'fibeAir4800OduAirRxPower', '-OsqnU', 'CERAGON-NET-MIB-FIBEAIR-4800');
```
snmpwalk_cache_oid() or one of the other snmpwalk_* functions should be used instead of snmp_walk() as it formats data better.

laf replied:
You've not added the mib CERAGON-NET-MIB-FIBEAIR-4800 being used here.

laf commented on includes/polling/os/ceragon.inc.php:
```diff
@@ -0,0 +1,6 @@
+<?php
+
+$hardware = trim(snmp_get($device, '1.3.6.1.4.1.4458.1000.1.1.30.0', '-OQv', '', ''), '"');
```
please convert these to a snmp_multi_get_oid() call so we can save on snmp lookups.

laf commented on includes/polling/mib/ceragon-mib.inc.php:
```diff
@@ -0,0 +1,96 @@
+<?php
```
We've recently added support for wireless sensors. this looks like rx/tx rate which isn't currently supported but it makes more sense to wait for that to be added to the core code than adding this old work around and us removing it later.

@murrant any thoughts?

laf commented on includes/definitions/ceragon.yaml:
```diff
@@ -0,0 +1,16 @@
+over:
+    - { graph: device_bits, text: 'Device Traffic' }
+    - { graph: device_ping_perf, text: 'Ping Times' }
+mib_dir:
```
You don't seem to have added any MIBS so you don't need this.

laf commented:
This also needs a test unit file: http://docs.librenms.org/Developing/os/Test-Units/

tamikkelsen commented:
Thank you for your feedback. 
This is my first attempt at adding a new device and it worked here so I thought I should add it in hopes of getting this kind of feedback so I could improve it. 

When it comes to the mib file, I added it when uploading but just as a file not as a git commit (was unsure of how I should  do it) I also did not commit the test file (will do that from now on:)) 

I will implement the feedback you gave, and also rename the OS from ceragon to fibeair as that is a much more informative name yes. 

With regards to the product; The FibeAir 2000 (or FA2000) is a product family, consisting of FA2000C, C+ and D+. Its the exact same products as RADWIN 2000 family - just rebranded. So with some small changes this should apply to RADWIN 2000 family aswell. some OIDs are different as far as i know - but I dont have any of the hardware to test on. 

Will update the files when I am done with the changes.

murrant commented:
@tamikkelsen You can find docs for the new wireless sensors here:
http://docs.librenms.org/Developing/os/Wireless-Sensors/

There are some types missing from that doc, I hope to get them added today.  You can look in LibreNMS/Interfaces/Discovery/Sensors in the meantime.

laf commented:
@tamikkelsen How are you getting on with the changes?

tamikkelsen commented:
I have completed most of the work, i will publish it tomorrow if i get the time. In working not perfect condition that is ;-)

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/42c4a0f8-7247-49d1-80c4-96c076c14884) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/42c4a0f8-7247-49d1-80c4-96c076c14884) completed: **4 new issues**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/86743ff1-3650-4a7e-b606-45859582e037) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/86743ff1-3650-4a7e-b606-45859582e037) completed: **4 new issues, 11 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/1c321038-424a-4c47-89a0-182d5878542c) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/1c321038-424a-4c47-89a0-182d5878542c) completed: **1 new issues, 11 updated code elements**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/048cda07-280c-41a4-8be3-8ce4442ac10b) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/048cda07-280c-41a4-8be3-8ce4442ac10b) completed: **11 updated code elements**

tamikkelsen commented:
I commited the files now, but for some reason it failes the Travis CI test. It failes on RADWIN test which has not been touched. But CERAGON radios are basically rebranded RADWIN radios just with some different OIDs. Anyone able to help me understand this?

laf commented:
You've edited the radwin.yaml file hence the Travis failure now.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/2f5b1fd3-b6f5-48a4-b8ef-47ad771b62a0) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/librenms/librenms/inspections/2f5b1fd3-b6f5-48a4-b8ef-47ad771b62a0) completed: **11 updated code elements**

laf commented:
You really need to stop using your master branch to develop in. You're getting more and more irrelevant commits now :(

http://docs.librenms.org/Developing/Using-Git/

tamikkelsen commented:
I’m so sorry. No GIT expert here. Will change this at once. 

 

Still don’t understand why it failes, will continue testing it without pushing it all to the master branch. 

 

 

 

Fra: Neil Lathwood <notifications@github.com>
Svar til: librenms/librenms <reply@reply.github.com>
Dato: lørdag 20. mai 2017 00.26
Til: librenms/librenms <librenms@noreply.github.com>
Kopi: Tommy Mikkelsen <tommy@mikkelsen.co>, Mention <mention@noreply.github.com>
Emne: Re: [librenms/librenms] newdevice: Added preliminary support for Ceragon FA2000 radios. (#6601)

 

You really need to stop using your master branch to develop in. You're getting more and more irrelevant commits now :(

http://docs.librenms.org/Developing/Using-Git/

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

tamikkelsen commented:
I still have huge problems getting the test to no fail. Any help would be appreciated.

 

What I did now, I removed all the changes I have done, did a git clean of my repo. Added in the needed files, for OS detection, poller, mibs etc. I have testet that it works as expected, but when running the test I get:

 

1) LibreNMS\Tests\OSDiscoveryTest::testAllFilesTested

Failed asserting that two arrays are equal.

--- Expected

+++ Actual

@@ @@

 Array (

-    0 => 'skel'

+    0 => 'fibeair'

+    1 => 'skel'

 )

 

/opt/librenms/tests/OSDiscoveryTest.php:92

 

I have tried adjusting both the snmprec files and the definition file to more specifically set the device apart, but same result when testing yet when running a discovery it works as expected. 

 

If anyone of you guys have any good ideas it sure would be appreciated.

 

 

Fra: Neil Lathwood <notifications@github.com>
Svar til: librenms/librenms <reply@reply.github.com>
Dato: lørdag 20. mai 2017 00.26
Til: librenms/librenms <librenms@noreply.github.com>
Kopi: Tommy Mikkelsen <tommy@mikkelsen.co>, Mention <mention@noreply.github.com>
Emne: Re: [librenms/librenms] newdevice: Added preliminary support for Ceragon FA2000 radios. (#6601)

 

You really need to stop using your master branch to develop in. You're getting more and more irrelevant commits now :(

http://docs.librenms.org/Developing/Using-Git/

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

laf commented:
I'll be honest and say you should close this and re-submit the changes as separate pull requests for each device. Jump on irc if you need help.
